lib/api.js

```diff
@@ -115,9 +130,20 @@ function makeApiRequest(transport, opts, bodyObj, callback) {
     });
   });
 
+  var originalArguments = arguments;
   req.on('error', function (err) {
-    logger.error('Could not make request to rollbar, ' + err);
-    callback(err);
+    // If the request to Rollbar failed due to a connection error, lets queue
+    // up the requests and try again periodically in the hopes that the connection
+    // will be restored.
+    //
+    // If no retryInterval set, interpret that as not wanting to retry send errors,
+    // so just bail out silently.
+    if (SETTINGS.retryInterval && err.code == 'ENOTFOUND') {
```

Comment:
Is this is the only error we want to retry on? I can't even find the existence of this documented anywhere but it does show up in Stack Overflow a lot as one of the errors. What about things like `ENETDOWN`? Does Node just return `ENOTFOUND` for all connection issues? I'm not that familiar with node specific network errors.